src/main/java/science/icebreaker/exception/AppExceptionHandler.java

```diff
@@ -0,0 +1,50 @@
+package science.icebreaker.exception;
+
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.*;
+
+@ControllerAdvice
+public class AppExceptionHandler {
+
+    private final MessageSource messageSource;
+
+    public AppExceptionHandler(MessageSource messageSource) {
+        this.messageSource = messageSource;
+    }
+
+    @ExceptionHandler({
+            AccountCreationException.class,
+            AccountNotFoundException.class,
+            InvalidFiltersException.class,
+            DeviceAvailabilityCreationException.class,
+            IllegalRequestParameterException.class
+    })
+    public ResponseEntity<Object> handleAccountCreationException(HttpServletRequest request, BaseException exception, Locale locale) {
```

Comment:
Because you are handling different exceptions here it would be better to choose a more general method name.

Reply:
@dajenet good idea choosing more general name, or handle different type of exceptions in different methods, but as method body is same thats why I put them in same. Feel free to change it as you consider.